src/LibXmlErrorGuard.cpp

```diff
@@ -0,0 +1,42 @@
+#include "LibXmlErrorGuard.h"
+extern "C" {
+    #include <libxml2/libxml/xmlerror.h>
+    #include <stdarg.h>
+}
+
+LibXmlErrorGuard::LibXmlErrorGuard()
+{
+    xmlSetGenericErrorFunc(&errorMessage,(xmlGenericErrorFunc)(&LibXmlErrorGuard::libxmlErrorCallback));
+}
+
+LibXmlErrorGuard::~LibXmlErrorGuard()
+{
+    // set default libxml error handler
+    xmlSetGenericErrorFunc(NULL, (xmlGenericErrorFunc)NULL);
+}
+
+const QString& LibXmlErrorGuard::getMessage() const
+{
+   return errorMessage;
+}
+
+bool LibXmlErrorGuard::isEmpty()
+{
+    return errorMessage.isEmpty();
+}
+
+void LibXmlErrorGuard::libxmlErrorCallback(QString* message, const char* format, ... )
+{
+    try
+    {
+        va_list args;
+        va_start(args, format);
+        QString newMessage = QString().vsprintf(format, args);
+        va_end(args);
+        message->append(newMessage);
+    }
+    catch(...)
+    {
+        ; // Possible exceptions aren't safe in C callback
```

Comment:
I don't see how exceptions can happen in the block, maybe I am missing something. Qt doesn't throw.